pkg/apis/flowcontrol/v1alpha1/defaults.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/api/flowcontrol/v1alpha1"
+	"k8s.io/kubernetes/pkg/apis/flowcontrol"
+)
+
+// SetDefaults_FlowSchema sets default values for flow schema
+func SetDefaults_FlowSchema(obj *v1alpha1.FlowSchema) {
+	if obj.Spec.MatchingPrecedence == 0 {
+		obj.Spec.MatchingPrecedence = flowcontrol.FlowSchemaDefaultMatchingPrecedence
+	}
+}
+
+// SetDefaults_FlowSchema sets default values for flow schema
+func SetDefaults_PriorityLevelConfiguration(obj *v1alpha1.PriorityLevelConfiguration) {
+	if !obj.Spec.Exempt {
+		if obj.Spec.HandSize == 0 {
```

Comment:
I have a slight preference for this be done only if the user doesn't specify a hand size. If they specify it and they said "0", we should give them an error. This is a strong suggestion but not mandatory, I guess.

Reply:
I am not sure I understand.  There is no difference between zero and not specified for this field.

Reply:
Yes, I'm suggesting that maybe there should be (i.e., make it a pointer).

Reply:
What's the point of introducing the ability to make a distinction between two specifications if one is to be rejected as an error?

Reply:
i remember we discussed about that all the way in the very first model pull, but ends up failing to convince anyone. that makes it a non-pointer field currently.